pkg/leaderelection/candidate_test.go

```diff
@@ -52,28 +47,23 @@ func TestLeaderElectionRun(t *testing.T) {
 
 	ctx := context.Background()
 
-	registryMock.EXPECT().Register(gomock.Any(), "leaderelection-test", 30*time.Second).Return("id", nil).Times(1)
-	registryMock.EXPECT().IsRegistered(gomock.Any(), "id").Return(true).AnyTimes()
-	registryMock.EXPECT().Deregister(gomock.Any(), "id").Return(nil).Times(1)
-	registryMock.EXPECT().Acquire(gomock.Any(), "id", common.GetBasePrefix()+"nodes/node01", gomock.Any()).Return(true, nil).AnyTimes()
 	registryMock.EXPECT().Acquire(gomock.Any(), "id", common.GetBasePrefix()+"leader/election/test", gomock.Any()).Return(true, nil).AnyTimes()
-	registryMock.EXPECT().RenewPeriodic(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil)
+	registryMock.EXPECT().Release(gomock.Any(), "id", common.GetBasePrefix()+"leader/election/test", gomock.Any()).Return(true).AnyTimes()
 	registryMock.EXPECT().Watch(gomock.Any(), gomock.Any()).Return(watcherMock, nil).AnyTimes()
 
 	config := getConfig()
 
 	config.Callbacks.OnStartedLeading = func(ctx context.Context) error {
 		return fmt.Errorf("terminate leader election")
 	}
-	model := getDummyNodeModel()
-	c, _ := New(model, config, registryMock)
+	c, _ := New("node01", "id", config, registryMock)
```

Comment:
Instead of ignoring we can `assert.Nil` on `err` here

Reply:
👍